src/core.c/io_operators.rakumod

```diff
@@ -282,34 +282,68 @@ multi sub indir(IO() $path, &what, :$d = True, :$r, :$w, :$x) {
     }
 }
 
-proto sub chmod($, |) {*}
-multi sub chmod($mode, *@filenames) {
+proto sub chmod($, |) is revision-gated("6.c") {*}
+multi sub chmod($mode, IO() $path) is revision-gated("6.c") { $path.chmod($mode) }
```

Comment:
Hmm, why is this candidate needed? I expected that I could delete this candidate and that the `multi sub chmod($mode, *@filenames) is revision-gated("6.c")` candidate would get used, but it doesn't:
```
raku t/spec/S16-filehandles/chmod.t
1..19
# Using file temp_1735619232
Cannot resolve caller chmod(Int:D, Str:D); none of these signatures matches:
    ($mode, IO(Any) $path)
    (IO(Any) $path, Int(Any) :$mode!)
    ($mode, *@filenames)
    ($mode, *@filenames)
  in block <unit> at t/spec/S16-filehandles/chmod.t line 38

# You planned 19 tests, but ran 0
```